### PR TITLE
Upgrade Astro to version 3

### PR DESCRIPTION
**Pull request by nickrttn**
Taking it one major at a time. This PR:

- upgrades Astro  to v3 and Astro plugins to versions matching Astro v3.
- updates the config accordingly
- removes a script that was used when migrating from Jekyll to Astro and will never be used again
- sets the default formatter for Astro files to the VS Code Astro plugin
- removes `remark-emoji`, it was only used for two emojis. Instead, force emoji's to color emoji's using system-default color emoji font families where needed, or use Unicode emoji's.
  - this package also has mismatching types with the upgraded Astro, and so removing it was the easiest route considering it was already fairly heavy-handed for the two emoji's we were using it for.
- fixes linting errors that were the result of the upgrade (eg. adding `is:inline` to some script tags)

### CODE REVIEW

File: src/pages/demo.mdx [hunk 1]
```diff
@@ -19,9 +20,9 @@ your files to our public [tusd](https://github.com/tus/tusd) instance, which is
 the official reference implementation for the tus protocol, written in the
 [Go programming language](https://golang.org/).
 
-**:warning: Please be aware that the uploaded files will be removed from our
-servers after a few hours, for privacy, and to combat people uploading malicious
-content.**
+**<Emoji label="warning">⚠️</Emoji> Please be aware that the uploaded files will
```
Comment (Acconut) [inline src/pages/demo.mdx:23]: Should all emojis be wrapped in this element now?

**Review by Acconut — APPROVED** [review]
Thank you very, very much for this PR!

File: .vscode/settings.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {
   "eslint.validate": ["astro"],
   "[astro]": {
-    "editor.defaultFormatter": "esbenp.prettier-vscode",
+    "editor.defaultFormatter": "astro-build.astro-vscode",
```
Comment (kvz) [inline .vscode/settings.json:4]: Should we add a `.vscode/extensions.json` with a `recommendations` object so this extension is proposed to be installed automatically?

File: src/content/protocols/1.0.x.md [hunk 3]
```diff
@@ -366,6 +366,10 @@ responses to `HEAD` requests.
 If the Server supports deferring length, it MUST add `creation-defer-length` to
 the `Tus-Extension` header.
 
+The `Upload-Length` header MAY be set to 0, indicating that the Client wants to
```
Comment (kvz) [inline src/content/protocols/1.0.x.md:369]: Unrelated?

File: src/pages/faq.md [hunk 4]
```diff
@@ -266,4 +266,4 @@ still an abbreviation) to desperately avoid yelling TUS in all caps. We used to
 insist on lowercasing, but have gotten more pragmatic from our encounters with
 journalists, who insist on it being Tus in their articles, and there's not much
 we can do about it. So now we just say: write it however you feel like that day
-:smile:
+😄
```
Comment (kvz) [inline src/pages/faq.md:269]: Does this need to be wrapped in an `<Emoji>` tag?

**Review by kvz — APPROVED** [review]
Awesome Nick! Just a few minor questions about this

File: .vscode/settings.json (hunk 2 above)
Comment (nickrttn) [reply .vscode/settings.json:4]: Already there ;)

File: src/pages/faq.md (hunk 4 above)
Comment (nickrttn) [reply src/pages/faq.md:269]: It doesn't need to be (but we can do it "just to be safe"). This emoji doesn't have a non-color icon equivalent in most fonts and will always show up as the color emoji defined in the font for the users' platform.

The ⚠️ emoji does have such a non-color equivalent, and so we force-set it to one of those color emoji system fonts.

**Comment by Acconut** [comment]
Thank you very much, Nick!
